tests/isolation_provider/test_qubes.py

```diff
@@ -0,0 +1,117 @@
+class TestQubes(TestIsolationProvider):
+    def test_max_pages_client_side_enforcement(
+        self,
+        sample_doc: str,
+        provider: Qubes,
+        mocker: MockerFixture,
+    ) -> None:
+        provider.progress_callback = mocker.MagicMock()
+        mocker.patch(
+            "dangerzone.conversion.errors.MAX_PAGES", 1
+        )  # sample_doc has 4 pages > 1
+        doc = Document(sample_doc)
+        with pytest.raises(errors.MaxPagesException):
+            success = provider._convert(doc, ocr_lang=None)
+            assert not success
+
+    def test_max_dimentions(
+        self,
+        sample_bad_width: str,
+        sample_bad_height: str,
+        provider: Qubes,
+        mocker: MockerFixture,
+    ) -> None:
+        provider.progress_callback = mocker.MagicMock()
+        with pytest.raises(errors.MaxPageWidthException):
+            success = provider._convert(Document(sample_bad_width), ocr_lang=None)
+            assert not success
+        with pytest.raises(errors.MaxPageHeightException):
+            success = provider._convert(Document(sample_bad_height), ocr_lang=None)
+            assert not success
+
+    def test_inconsistent_dimentions(
+        self,
+        provider: Qubes,
+        mocker: MockerFixture,
+        monkeypatch: MonkeyPatch,
+        sample_doc: str,
+    ) -> None:
+        provider.progress_callback = mocker.MagicMock()
+
+        def qrexec_subprocess() -> subprocess.Popen:
+            p = subprocess.Popen(
+                ["tee"],  # pipe out whatever we feed into stdin
+                stdin=subprocess.PIPE,
+                stdout=subprocess.PIPE,
+                stderr=subprocess.PIPE,
+            )
+            assert p.stdin is not None
+            p.stdin.write(b"\x00\x01")  # 1 page
+            p.stdin.write(b"\x00\x04")  # 4px width
+            p.stdin.write(b"\x00\x05")  # 5px height
+            p.stdin.write(b"\x00")  # only one byte received but 3*4*5 expected
+            return p
+
+        monkeypatch.setattr(provider, "qrexec_subprocess", qrexec_subprocess)
+        with pytest.raises(errors.InterruptedConversion) as e:
+            doc = Document(sample_doc)
+            provider._convert(doc, ocr_lang=None)
+
+    def test_out_of_ram(
+        self,
+        provider: Qubes,
+        mocker: MockerFixture,
+        monkeypatch: MonkeyPatch,
+        sample_doc: str,
+    ) -> None:
+        provider.progress_callback = mocker.MagicMock()
+
+        def qrexec_subprocess() -> subprocess.Popen:
+            p = subprocess.Popen(
+                # XXX error 126 simulates a qrexec-policy failure. Source:
+                # https://github.com/QubesOS/qubes-core-qrexec/blob/fdcbfd7/daemon/qrexec-daemon.c#L1022
+                ["exit 126"],
+                stdin=subprocess.PIPE,
+                stdout=subprocess.PIPE,
+                stderr=subprocess.PIPE,
+                shell=True,
+            )
+            time.sleep(0.2)  # Avoid race condition
```

Comment:
What is the race condition that we are avoiding with this sleep argument?

Reply:
It was an old comment. Thanks for catching it! Fixed in be0fbf21.